src/init.cpp

```diff
@@ -1361,217 +1363,100 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
-                    });
-
-                    // If necessary, upgrade from older database format.
-                    // This is a no-op if we cleared the coinsviewdb with -reindex or -reindex-chainstate
-                    if (!chainstate->CoinsDB().Upgrade()) {
-                        strLoadError = _("Error upgrading chainstate database");
-                        failed_chainstate_init = true;
-                        break;
-                    }
-
-                    // ReplayBlocks is a no-op if we cleared the coinsviewdb with -reindex or -reindex-chainstate
-                    if (!chainstate->ReplayBlocks()) {
-                        strLoadError = _("Unable to replay blocks. You will need to rebuild the database using -reindex-chainstate.");
-                        failed_chainstate_init = true;
-                        break;
-                    }
-
-                    // The on-disk coinsdb is now in a good state, create the cache
-                    chainstate->InitCoinsCache(nCoinCacheUsage);
-                    assert(chainstate->CanFlushToDisk());
-
-                    if (!is_coinsview_empty(chainstate)) {
-                        // LoadChainTip initializes the chain based on CoinsTip()'s best block
-                        if (!chainstate->LoadChainTip()) {
-                            strLoadError = _("Error initializing block database");
-                            failed_chainstate_init = true;
-                            break; // out of the per-chainstate loop
-                        }
-                        assert(chainstate->m_chain.Tip() != nullptr);
-                    }
-                }
-
-                if (failed_chainstate_init) {
-                    break; // out of the chainstate activation do-while
-                }
-            } catch (const std::exception& e) {
-                LogPrintf("%s\n", e.what());
-                strLoadError = _("Error opening block database");
+        const int64_t load_block_index_start_time = GetTimeMillis();
+        std::optional<ChainstateLoadingError> rv;
+        try {
+            rv = LoadChainstateSequence(fReset,
+                                        chainman,
+                                        Assert(node.mempool.get()),
+                                        fPruneMode,
+                                        chainparams.GetConsensus(),
+                                        fReindexChainState,
+                                        cache_sizes.block_tree_db_cache_size,
+                                        cache_sizes.coin_db_cache_size,
+                                        cache_sizes.coin_cache_usage_size,
+                                        args.GetIntArg("-checkblocks", DEFAULT_CHECKBLOCKS),
+                                        args.GetIntArg("-checklevel", DEFAULT_CHECKLEVEL),
+                                        false,
+                                        false,
```

Comment:
Can you add inline comments for these arguments (that can be checked by a clang-tidy):

```suggestion
                                        /*block_tree_db_in_memory=*/false,
                                        /*coins_db_in_memory=*/false,
```